kotlin/internal/js/js.bzl

```diff
@@ -0,0 +1,63 @@
+load(
+    "//kotlin/internal:defs.bzl",
+    _KtJsInfo = "KtJsInfo",
+    _TOOLCHAIN_TYPE = "TOOLCHAIN_TYPE",
+)
+load(":impl.bzl", "kt_js_import_impl", "kt_js_library_impl")
+
+"""
+kt_js_import exposes .klib files to other kt_js_library targets
+"""
+kt_js_import = rule(
+    implementation = kt_js_import_impl,
+    attrs = {
+        "klibs": attr.label_list(
+            doc = "One or more klib targets that model a kotlin/js library",
+            allow_files = [".klib"],
+        ),
+    },
+    toolchains = [_TOOLCHAIN_TYPE],
+    provides = [_KtJsInfo],
+)
+
+"""
+kt_js_library runs the compilation of Kotlin to intermediate klib files to be shared
+with other kt_js_library targets or generates JS directly to then run directly with Node/Javascript
+runtim
+"""
+kt_js_library = rule(
+    implementation = kt_js_library_impl,
+    attrs = {
+        "srcs": attr.label_list(
+            doc = "A list of source files to be transpiled to JS",
+            allow_files = [".kt"],
+        ),
+        "deps": attr.label_list(
+            doc = "A list of other kt_js_library that this library depends on for compilation",
+            providers = [_KtJsInfo],
+        ),
+        "output_kind": attr.string(
```

Comment:
What is the rationale of changing the outputs via attributes (`output_kind`, `sourcemap`)? 

It's not a common pattern for bazel.

Reply:
It was added to avoid creating `.js` outputs on every `kt_js_library` target as it's only really required in the final target when running it as Node binary (as an example) - only klibs are required for sharing between targets for compilation. The JS creation part seemed quite slow so I made it conditional - I can remove it though and just have it generated by default if desired.